soh/src/code/z_message_PAL.c

```diff
@@ -2736,7 +2736,7 @@ void Message_DrawMain(GlobalContext* globalCtx, Gfx** p) {
                     if (msgCtx->lastPlayedSong < OCARINA_SONG_SARIAS &&
                         (msgCtx->ocarinaAction < OCARINA_ACTION_PLAYBACK_MINUET ||
                          msgCtx->ocarinaAction >= OCARINA_ACTION_PLAYBACK_SARIA)) {
-                        if (msgCtx->disableWarpSongs || interfaceCtx->restrictions.warpSongs == 3) {
+                        if (msgCtx->disableWarpSongs || (interfaceCtx->restrictions.warpSongs == 3 && !gSaveContext.n64ddFlag)) {
```

Comment:
do you know when warping is disabled by `msgCtx`? it's not clear why we want rando to override the `interfaceCtx` restrictions but not the `msgCtx` ones

Reply:
I do not, though the linked issue only mentioned GTG & Ganon’s Castle, reason being I didn’t dig much deeper into the msgCtx restrictions. I’ll look into that further if that blocks this from going in